pkg/config/config.go

```diff
@@ -0,0 +1,303 @@
+	// Image is the image to use in the helm step.
+	//
+	// There's no config value for the Docker image tag to use, as that comes
+	// from the helmVersion field in the helm step type from the .wharf-ci.yml
+	// file.
+	//
+	// Added in v0.8.0.
+	Image string
+}
+
+// ProvisionerConfig holds settings for the provisioner.
+type ProvisionerConfig struct {
+	HTTP   HTTPConfig
+	Worker ProvisionerWorkerConfig
+}
+
+// ProvisionerWorkerConfig holds settings for worker pods that are created by the
+// provisioner.
+type ProvisionerWorkerConfig struct {
+	// ServiceAccountName is the service account name to use for the pod.
+	//
+	// Added in v0.8.0.
+	ServiceAccountName string
+	// InitContainer holds settings for the init container.
+	//
+	// Added in v0.8.0.
+	InitContainer K8sContainerConfig
+	// Container holds settings for the container.
+	//
+	// Added in v0.8.0.
+	Container K8sContainerConfig
+}
+
+// K8sContainerConfig holds settings for a kubernetes container.
+type K8sContainerConfig struct {
+	// Image is the base path for the image to use for a container.
+	//
+	// Added in v0.8.0.
+	Image string
+	// ImageTag is the version tag to use for a container.
+	//
+	// Added in v0.8.0.
+	ImageTag string
+	// ImagePullPolicy is the image pull policy to use for a container.
+	//
+	// "Always"
+	//   Attempts to pull the latest image.
+	//   Container will fail if the pull fails.
+	//
+	// "Never"
+	//   Never pulls an image, i.e. only uses a local image.
+	//   Container will fail if the image isn't present
+	//
+	// "IfNotPresent"
+	//   Pulls if the image isn't present on disk.
+	//   Container will fail if the image isn't present and the pull fails.
+	//
+	// Added in v0.8.0.
+	ImagePullPolicy v1.PullPolicy
+}
+
+// HTTPConfig holds settings for the HTTP server.
+type HTTPConfig struct {
+	CORS CORSConfig
+	// BindAddress is the IP-address and port, separated by a colon, to bind
+	// the HTTP server to. An IP-address of 0.0.0.0 will bind to all
+	// IP-addresses.
+	//
+	// Added in v0.8.0.
+	BindAddress string
+}
+
+// CORSConfig holds settings for the HTTP server's CORS settings.
+type CORSConfig struct {
+	// AllowAllOrigins enables CORS and allows all hostnames and URLs in the
+	// HTTP request origins when set to true. Practically speaking, this
+	// results in the HTTP header "Access-Control-Allow-Origin" set to "*".
+	//
+	// Added in v0.8.0.
+	AllowAllOrigins bool
+	// AllowOrigins enables CORS and allows the list of origins in the
+	// HTTP request origins when set. Practically speaking, this
+	// results in the HTTP header "Access-Control-Allow-Origin".
+	//
+	// Added in v0.8.0.
+	AllowOrigins []string
+}
+
+// AggregatorConfig holds settings for the aggregator.
+type AggregatorConfig struct {
+	// WharfAPIURL is the URL used to connect to Wharf API.
+	//
+	// Added in v0.8.0.
+	WharfAPIURL string
+	// WorkerAPIExternalPort is the port used to connect to a Wharf worker.
+	//
+	// Added in v0.8.0.
+	WorkerAPIExternalPort string
```

Comment:
This should be of type `int16`, no? Or am I missing something? Can the wharf-core config package not parse int16s?

Reply:
Tested and works, changing 👍🏻